multiqc/utils/software_versions.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python
+
+""" Super Special-Case MultiQC module to produce report section on software versions """
+
+
+import logging
+
+import yaml
+
+from multiqc.modules.base_module import BaseMultiqcModule
+from multiqc.utils import report
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+
+class MultiqcModule(BaseMultiqcModule):
+    def __init__(self):
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(
+            name="Software Versions",
+            anchor="multiqc_software_versions",
+            info="Versions of software tools extracted from file contents.",
+        )
+
+        # Don't repeat the Custom Content name in the subtext
+        if self.info or self.extra:
+            self.intro = "<p>{}</p>{}".format(self.info, self.extra)
+
+        self.report_software_versions()
+
+    def report_software_versions(self):
+        """Create section listing software versions."""
+        content = "<dl class=dl-horizontal>\n"
+        for tool_name in sorted(report.software_versions, key=lambda x: x.lower()):
```

Comment:
_/ * waits for someone to request a custom sort order on this * /_

Reply:
Semi serious about this. If we do write this so that people can for example list the processes in their pipeline with associated software versions, they will likely want to maintain the order.

Yet another config file option..? 🤔 As I think sorting makes sense most of the time..

Reply:
Meh, can't think of a good way to handle this for now. Let's cross the bridge when someone asks for it.